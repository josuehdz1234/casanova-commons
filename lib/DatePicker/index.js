"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _react = _interopRequireWildcard(require("react"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _momentTimezone = _interopRequireDefault(require("moment-timezone"));
var _classnames = _interopRequireDefault(require("classnames"));
var _formik = require("formik");
require("./DatePicker.scss");
var _Input = _interopRequireDefault(require("../Input/Input"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }
function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { _defineProperty(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }
function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }
function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }
function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }
function _iterableToArrayLimit(arr, i) { var _i = null == arr ? null : "undefined" != typeof Symbol && arr[Symbol.iterator] || arr["@@iterator"]; if (null != _i) { var _s, _e, _x, _r, _arr = [], _n = !0, _d = !1; try { if (_x = (_i = _i.call(arr)).next, 0 === i) { if (Object(_i) !== _i) return; _n = !1; } else for (; !(_n = (_s = _x.call(_i)).done) && (_arr.push(_s.value), _arr.length !== i); _n = !0); } catch (err) { _d = !0, _e = err; } finally { try { if (!_n && null != _i["return"] && (_r = _i["return"](), Object(_r) !== _r)) return; } finally { if (_d) throw _e; } } return _arr; } }
function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }
var YYYY_MM_DD = 'YYYY-MM-DD';
var timeZone = 'America/Mexico_City';
var DatePicker = function DatePicker(props) {
  var initialValue = props.initialValue,
    disabled = props.disabled,
    name = props.name,
    onChange = props.onChange,
    calendarTop = props.calendarTop,
    _props$initialYear = props.initialYear,
    initialYear = _props$initialYear === void 0 ? new Date().getFullYear() : _props$initialYear,
    _props$numberYears = props.numberYears,
    numberYears = _props$numberYears === void 0 ? 12 : _props$numberYears;
  var _useField = (0, _formik.useField)(name),
    _useField2 = _slicedToArray(_useField, 3),
    field = _useField2[0],
    meta = _useField2[1],
    helpers = _useField2[2];
  var calendarRef = (0, _react.useRef)(null);
  var setValue = helpers.setValue;
  var days = (0, _react.useMemo)(function () {
    return Array.from({
      length: 31
    }, function (_, i) {
      return i + 1;
    });
  });
  var months = (0, _react.useMemo)(function () {
    return ['Ene', 'Feb', 'Mar', 'Abr', 'May', 'Jun', 'Jul', 'Ago', 'Sep', 'Oct', 'Nov', 'Dic'];
  });
  var years = (0, _react.useMemo)(function () {
    return Array.from({
      length: numberYears
    }, function (_, i) {
      return initialYear + 1 - i;
    });
  });
  var _useState = (0, _react.useState)('day'),
    _useState2 = _slicedToArray(_useState, 2),
    activeSelection = _useState2[0],
    setActiveSelection = _useState2[1];
  var _useState3 = (0, _react.useState)('día'),
    _useState4 = _slicedToArray(_useState3, 2),
    daySelected = _useState4[0],
    setDaySelected = _useState4[1];
  var _useState5 = (0, _react.useState)('mes'),
    _useState6 = _slicedToArray(_useState5, 2),
    monthSelected = _useState6[0],
    setMonthSelected = _useState6[1];
  var _useState7 = (0, _react.useState)('año'),
    _useState8 = _slicedToArray(_useState7, 2),
    yearSelected = _useState8[0],
    setYearSelected = _useState8[1];
  var _useState9 = (0, _react.useState)(days),
    _useState10 = _slicedToArray(_useState9, 2),
    calendarElements = _useState10[0],
    setCalendarElements = _useState10[1];
  var _useState11 = (0, _react.useState)('día'),
    _useState12 = _slicedToArray(_useState11, 2),
    currentTitle = _useState12[0],
    setCurrentTitle = _useState12[1];
  var _useState13 = (0, _react.useState)(false),
    _useState14 = _slicedToArray(_useState13, 2),
    open = _useState14[0],
    setOpen = _useState14[1];
  var _useState15 = (0, _react.useState)(false),
    _useState16 = _slicedToArray(_useState15, 2),
    hasContent = _useState16[0],
    setHasContent = _useState16[1];
  var handleSelection = (0, _react.useCallback)(function (value) {
    if (activeSelection === 'day') {
      setDaySelected(value);
      setActiveSelection('month');
      setCalendarElements(months);
      setCurrentTitle('mes');
    } else if (activeSelection === 'month') {
      setMonthSelected(months.indexOf(value));
      setActiveSelection('year');
      setCalendarElements(years);
      setCurrentTitle('año');
    } else {
      setYearSelected(value);
      setActiveSelection('day');
      setCalendarElements(days);
      setCurrentTitle('día');
      setOpen(false);
    }
  }, [activeSelection, years, days, months]);
  (0, _react.useEffect)(function () {
    if (daySelected !== 'día' || monthSelected !== 'mes' || yearSelected !== 'año') {
      setHasContent(true);
    } else {
      setHasContent(false);
    }
    if (daySelected !== 'día' && monthSelected !== 'mes' && yearSelected !== 'año') {
      var realMonth = monthSelected + 1;
      var month = realMonth >= 10 ? realMonth : "0".concat(realMonth);
      var day = daySelected >= 10 ? daySelected : "0".concat(daySelected);
      setValue("".concat(yearSelected, "-").concat(month, "-").concat(day));
    }
  }, [yearSelected, daySelected, monthSelected]);
  var toogleOpen = (0, _react.useCallback)(function () {
    setOpen(!open);
  }, [open]);
  var handleCalendarClick = (0, _react.useCallback)(function () {
    if (!disabled) {
      toogleOpen();
    }
  });
  var handleOpen = (0, _react.useCallback)(function () {
    setOpen(true);
  }, []);
  (0, _react.useEffect)(function () {
    if (initialValue) {
      var propsDate = (0, _momentTimezone["default"])(initialValue, YYYY_MM_DD).tz(timeZone).toDate();
      setDaySelected(propsDate.getDate());
      setMonthSelected(propsDate.getMonth());
      setYearSelected(propsDate.getFullYear());
    } else {
      setDaySelected('día');
      setMonthSelected('mes');
      setYearSelected('año');
    }
  }, [initialValue]);
  var handleBlur = (0, _react.useCallback)(function (e) {
    setTimeout(function () {
      var target = document.activeElement;
      var focusedElement = target ? target.id || target || target.tagName : '';
      var isFocusedCalendar = focusedElement.className.indexOf('DatePickerV1__calendar--open') !== -1;
      setOpen(isFocusedCalendar);
      if (props.onBlur) {
        props.onBlur(e);
      }
    }, 1);
  });
  var handleKeyDown = (0, _react.useCallback)(function (event) {
    event.preventDefault();
  }, []);
  (0, _react.useEffect)(function () {
    if (onChange) {
      onChange(field.value);
    }
  }, [field.value]);

  // Starts classNames block
  var containerCN = (0, _classnames["default"])('DatePickerV1__input', {
    'DatePickerV1__input--open': open
  });
  var calendarCN = (0, _classnames["default"])(calendarTop && 'DatePickerV1__calendarTop', "DatePickerV1__calendar", {
    'DatePickerV1__calendar--open': open
  });
  var iconCn = (0, _classnames["default"])("svg-icon svg-icon-calendar");
  var noValues = daySelected === 'día' && monthSelected === 'mes' && yearSelected === 'año';
  var dayFormatted = daySelected === 'día' || daySelected >= 10 ? daySelected : "0".concat(daySelected);
  var monthFormatted = months[monthSelected] ? months[monthSelected] : 'mes';
  var finalValue = noValues ? '' : "".concat(dayFormatted, " / ").concat(monthFormatted, " / ").concat(yearSelected);
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: "DatePickerV1"
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: containerCN
  }, /*#__PURE__*/_react["default"].createElement(_Input["default"], _extends({
    label: props.label
  }, field, {
    type: "text",
    value: finalValue,
    placeholder: props.placeholder,
    modifiedExternal: hasContent,
    name: props.name,
    id: props.id,
    disabled: disabled,
    onBlur: handleBlur,
    onKeyDown: handleKeyDown,
    right: /*#__PURE__*/_react["default"].createElement("div", {
      onBlur: handleBlur,
      onClick: handleCalendarClick,
      tabIndex: "-1",
      className: iconCn
    }),
    onFocus: handleOpen,
    inputClass: calendarTop
  }))), /*#__PURE__*/_react["default"].createElement("div", {
    tabIndex: -1,
    ref: calendarRef,
    onBlur: function onBlur(e) {
      setOpen(false);
      if (props.onBlur) {
        props.onBlur(_objectSpread(_objectSpread({}, e), {}, {
          dateValue: field.value
        }));
      }
    },
    className: calendarCN
  }, /*#__PURE__*/_react["default"].createElement("label", {
    className: "col-form-label"
  }, "Selecciona un ", currentTitle), /*#__PURE__*/_react["default"].createElement("div", {
    className: "DatePickerV1__calendar__".concat(activeSelection, "s-grid")
  }, calendarElements && calendarElements.map(function (element) {
    return /*#__PURE__*/_react["default"].createElement("span", {
      key: "".concat(activeSelection, "-").concat(element),
      onClick: function onClick() {
        return handleSelection(element);
      },
      className: (0, _classnames["default"])('DatePickerV1__calendar__element', {
        'DatePickerV1__calendar__element--selected': element === daySelected || element === months[monthSelected] || element === yearSelected
      })
    }, activeSelection === 'day' ? element >= 10 ? element : "0".concat(element) : element);
  }))));
};
DatePicker.propTypes = {
  label: _propTypes["default"].string,
  name: _propTypes["default"].string,
  placeholder: _propTypes["default"].string
};
DatePicker.defaultProps = {};
var _default = DatePicker;
exports["default"] = _default;