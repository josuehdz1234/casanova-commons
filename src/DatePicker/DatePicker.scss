.DatePickerV1 {
  width: 100%;
  position: relative;
  &__input {
    &--open {
      cursor: pointer;
      div.form-group {
        margin-bottom: 0;
      }
      input.form-control {
        border: 2px solid #eceef4;
        border-radius: 0.8rem 0.8rem 0 0;
        border-bottom: solid 0;
        background-color: white;
        &:focus {
          box-shadow: none;
        }
      }
      input.formtopPosition {
        border: 2px solid #eceef4;
        border-radius: 0 0 0.8rem 0.8rem!important;
        background-color: white;
        &:focus {
          box-shadow: none;
        }
      }
    }
    &--with-content:not(:disabled):not([type="number"]) {
      input.form-control{
        background-color: white;
      }
      .col-form-label{
        visibility: visible;
      }
    }
  }
  &__calendar {
    width: 100%;
    height: 16rem;
    border: 2px solid #eceef4;
    position: absolute;
    background-color: white;
    align-items: center;
    flex-direction: column;
    z-index: 10;
    display: none;
    overflow-y: scroll;
    &--open {
      display: flex;
    }

    &__element {
      font-weight: 900;
      border-right: solid 0;
      cursor: pointer;
      padding: 0.5rem 0 0 0;
      text-align: center;
      white-space: nowrap;
      border: none;
      color: #007aff;
      &--selected {
        font-weight: normal;
        color: black;
      }
    }
    &__days-grid {
      display: grid;
      grid-template-columns: repeat(6, 2.5rem);
      grid-template-rows: repeat(5, 2rem);
    }
    &__months-grid {
      display: grid;
      grid-template-columns: repeat(3, 6rem);
      grid-template-rows: repeat(4, 3rem);
      .page-link {
        line-height: 2rem;
      }
    }
    &__years-grid {
      display: grid;
      grid-template-columns: repeat(3, 6rem);
      grid-auto-rows: 3rem;
      .page-link {
        line-height: 2rem;
      }
    }
  }
  &__calendarTop {
    top: -220px;
  }
}
