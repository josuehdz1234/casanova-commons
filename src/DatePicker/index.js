import React, {
  useState,
  useCallback,
  useEffect,
  useMemo,
  useRef,
} from 'react';
import PropTypes from 'prop-types';
import moment from 'moment-timezone';
import classNames from 'classnames';
import { useField } from 'formik';
import './DatePicker.scss';

import Input from '../Input/Input';

const YYYY_MM_DD = 'YYYY-MM-DD';
const timeZone = 'America/Mexico_City';

const DatePicker = (props) => {
  const {
    initialValue,
    disabled,
    name,
    onChange,
    calendarTop,
    initialYear = new Date().getFullYear(),
    numberYears = 12,
  } = props;
  const [field, meta, helpers] = useField(name);
  const calendarRef = useRef(null);

  const { setValue } = helpers;

  const days = useMemo(() => Array.from({ length: 31 }, (_, i) => i + 1));
  const months = useMemo(() => [
    'Ene',
    'Feb',
    'Mar',
    'Abr',
    'May',
    'Jun',
    'Jul',
    'Ago',
    'Sep',
    'Oct',
    'Nov',
    'Dic',
  ]);
  const years = useMemo(() =>
    Array.from({ length: numberYears }, (_, i) => initialYear + 1 - i)
  );

  const [activeSelection, setActiveSelection] = useState('day');
  const [daySelected, setDaySelected] = useState('día');
  const [monthSelected, setMonthSelected] = useState('mes');
  const [yearSelected, setYearSelected] = useState('año');
  const [calendarElements, setCalendarElements] = useState(days);
  const [currentTitle, setCurrentTitle] = useState('día');
  const [open, setOpen] = useState(false);
  const [hasContent, setHasContent] = useState(false);

  const handleSelection = useCallback(
    (value) => {
      if (activeSelection === 'day') {
        setDaySelected(value);
        setActiveSelection('month');
        setCalendarElements(months);
        setCurrentTitle('mes');
      } else if (activeSelection === 'month') {
        setMonthSelected(months.indexOf(value));
        setActiveSelection('year');
        setCalendarElements(years);
        setCurrentTitle('año');
      } else {
        setYearSelected(value);
        setActiveSelection('day');
        setCalendarElements(days);
        setCurrentTitle('día');
        setOpen(false);
      }
    },
    [activeSelection, years, days, months]
  );

  useEffect(() => {
    if (
      daySelected !== 'día' ||
      monthSelected !== 'mes' ||
      yearSelected !== 'año'
    ) {
      setHasContent(true);
    } else {
      setHasContent(false);
    }
    if (
      daySelected !== 'día' &&
      monthSelected !== 'mes' &&
      yearSelected !== 'año'
    ) {
      const realMonth = monthSelected + 1;
      const month = realMonth >= 10 ? realMonth : `0${realMonth}`;
      const day = daySelected >= 10 ? daySelected : `0${daySelected}`;
      setValue(`${yearSelected}-${month}-${day}`);
    }
  }, [yearSelected, daySelected, monthSelected]);

  const toogleOpen = useCallback(() => {
    setOpen(!open);
  }, [open]);

  const handleCalendarClick = useCallback(() => {
    if (!disabled) {
      toogleOpen();
    }
  });

  const handleOpen = useCallback(() => {
    setOpen(true);
  }, []);

  useEffect(() => {
    if (initialValue) {
      const propsDate = moment(initialValue, YYYY_MM_DD).tz(timeZone).toDate();
      setDaySelected(propsDate.getDate());
      setMonthSelected(propsDate.getMonth());
      setYearSelected(propsDate.getFullYear());
    } else {
      setDaySelected('día');
      setMonthSelected('mes');
      setYearSelected('año');
    }
  }, [initialValue]);

  const handleBlur = useCallback((e) => {
    setTimeout(() => {
      const target = document.activeElement;
      const focusedElement = target
        ? target.id || target || target.tagName
        : '';
      const isFocusedCalendar =
        focusedElement.className.indexOf('DatePickerV1__calendar--open') !== -1;
      setOpen(isFocusedCalendar);
      if (props.onBlur) {
        props.onBlur(e);
      }
    }, 1);
  });

  const handleKeyDown = useCallback((event) => {
    event.preventDefault();
  }, []);

  useEffect(() => {
    if (onChange) {
      onChange(field.value);
    }
  }, [field.value]);

  // Starts classNames block
  const containerCN = classNames('DatePickerV1__input', {
    'DatePickerV1__input--open': open,
  });
  const calendarCN = classNames(
    calendarTop && 'DatePickerV1__calendarTop',
    `DatePickerV1__calendar`,
    { 'DatePickerV1__calendar--open': open }
  );
  const iconCn = classNames(`svg-icon svg-icon-calendar`);

  const noValues =
    daySelected === 'día' && monthSelected === 'mes' && yearSelected === 'año';
  const dayFormatted =
    daySelected === 'día' || daySelected >= 10
      ? daySelected
      : `0${daySelected}`;
  const monthFormatted = months[monthSelected] ? months[monthSelected] : 'mes';
  const finalValue = noValues
    ? ''
    : `${dayFormatted} / ${monthFormatted} / ${yearSelected}`;

  return (
    <div className="DatePickerV1">
      <div className={containerCN}>
        <Input
          label={props.label}
          {...field}
          type="text"
          value={finalValue}
          placeholder={props.placeholder}
          modifiedExternal={hasContent}
          name={props.name}
          id={props.id}
          disabled={disabled}
          onBlur={handleBlur}
          onKeyDown={handleKeyDown}
          right={
            <div
              onBlur={handleBlur}
              onClick={handleCalendarClick}
              tabIndex="-1"
              className={iconCn}
            />
          }
          onFocus={handleOpen}
          inputClass={calendarTop}
        />
      </div>
      <div
        tabIndex={-1}
        ref={calendarRef}
        onBlur={(e) => {
          setOpen(false);
          if (props.onBlur) {
            props.onBlur({
              ...e,
              dateValue: field.value,
            });
          }
        }}
        className={calendarCN}
      >
        <label className="col-form-label">Selecciona un {currentTitle}</label>
        <div className={`DatePickerV1__calendar__${activeSelection}s-grid`}>
          {calendarElements &&
            calendarElements.map((element) => (
              <span
                key={`${activeSelection}-${element}`}
                onClick={() => handleSelection(element)}
                className={classNames('DatePickerV1__calendar__element', {
                  'DatePickerV1__calendar__element--selected':
                    element === daySelected ||
                    element === months[monthSelected] ||
                    element === yearSelected,
                })}
              >
                {activeSelection === 'day'
                  ? element >= 10
                    ? element
                    : `0${element}`
                  : element}
              </span>
            ))}
        </div>
      </div>
    </div>
  );
};

DatePicker.propTypes = {
  label: PropTypes.string,
  name: PropTypes.string,
  placeholder: PropTypes.string,
};

DatePicker.defaultProps = {};

export default DatePicker;
