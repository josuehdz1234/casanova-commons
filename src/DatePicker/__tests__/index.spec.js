import React from 'react';
import { mount } from 'enzyme';
import { Formik } from 'formik';
import { render, fireEvent } from '@testing-library/react';

import DatePicker from '..';

describe('DatePicker', () => {
  it('should render DatePicker with defaults', () => {
    const wrapper = render(
      <Formik>
        <DatePicker label="Fecha" name="name" />
      </Formik>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should have open class when focus', () => {
    const wrapper = mount(
      <Formik>
        <DatePicker label="Fecha" name="name" />
      </Formik>
    );
    const datePicker = wrapper.find('DatePicker');
    const datePickerInput = datePicker.find('.DatePicker__input');
    const datePickerCalendar = datePicker.find('.DatePicker__calendar');

    datePickerInput.find('input').simulate('focus');

    expect(datePickerCalendar.instance().className).toEqual(
      'DatePicker__calendar DatePicker__calendar--open'
    );
    expect(datePickerInput.instance().className).toEqual(
      'DatePicker__input DatePicker__input--open'
    );
  });

  it('should set value when click on month', () => {
    const { container } = render(
      <Formik>
        <DatePicker label="Fecha" name="name" />
      </Formik>
    );
    const datePickerCalendar = container.querySelector('.DatePicker__calendar');
    fireEvent.focus(container.querySelector('input'));

    let element = datePickerCalendar.querySelector(
      '.DatePicker__calendar__element'
    );
    fireEvent.click(element);

    expect(container.querySelector('input').value).toEqual('01 / mes / año');

    element = datePickerCalendar.querySelector(
      '.DatePicker__calendar__element'
    );
    fireEvent.click(element);

    expect(container.querySelector('input').value).toEqual('01 / Ene / año');

    // TODO: Handle Formik props inside component
    // element = datePickerCalendar.querySelector('.DatePicker__calendar__element');
    // fireEvent.click(element);

    // expect(container.querySelector('input').value).toEqual('01 / Ene / 2020');
  });
});
