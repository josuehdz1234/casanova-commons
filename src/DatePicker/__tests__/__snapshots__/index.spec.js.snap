// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker should render DatePicker with defaults 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="DatePicker"
      >
        <div
          class="DatePicker__input"
        >
          <div
            class="form-group  "
          >
            <label
              class="col-form-label"
              style="visibility: hidden;"
            >
              Fecha
            </label>
            <div
              class="custom-input-group"
            >
              <input
                class="undefined form-control input-group-br-no"
                type="text"
                value=""
              />
              <span
                class="right-content"
              >
                <div
                  class="svg-icon svg-icon-calendar"
                />
              </span>
            </div>
          </div>
        </div>
        <div
          class="DatePicker__calendar"
          tabindex="0"
        >
          <label
            class="col-form-label"
          >
            Selecciona un 
            día
          </label>
          <div
            class="DatePicker__calendar__days-grid"
          >
            <span
              class="DatePicker__calendar__element"
            >
              01
            </span>
            <span
              class="DatePicker__calendar__element"
            >
              02
            </span>
            <span
              class="DatePicker__calendar__element"
            >
              03
            </span>
            <span
              class="DatePicker__calendar__element"
            >
              04
            </span>
            <span
              class="DatePicker__calendar__element"
            >
              05
            </span>
            <span
              class="DatePicker__calendar__element"
            >
              06
            </span>
            <span
              class="DatePicker__calendar__element"
            >
              07
            </span>
            <span
              class="DatePicker__calendar__element"
            >
              08
            </span>
            <span
              class="DatePicker__calendar__element"
            >
              09
            </span>
            <span
              class="DatePicker__calendar__element"
            >
              10
            </span>
            <span
              class="DatePicker__calendar__element"
            >
              11
            </span>
            <span
              class="DatePicker__calendar__element"
            >
              12
            </span>
            <span
              class="DatePicker__calendar__element"
            >
              13
            </span>
            <span
              class="DatePicker__calendar__element"
            >
              14
            </span>
            <span
              class="DatePicker__calendar__element"
            >
              15
            </span>
            <span
              class="DatePicker__calendar__element"
            >
              16
            </span>
            <span
              class="DatePicker__calendar__element"
            >
              17
            </span>
            <span
              class="DatePicker__calendar__element"
            >
              18
            </span>
            <span
              class="DatePicker__calendar__element"
            >
              19
            </span>
            <span
              class="DatePicker__calendar__element"
            >
              20
            </span>
            <span
              class="DatePicker__calendar__element"
            >
              21
            </span>
            <span
              class="DatePicker__calendar__element"
            >
              22
            </span>
            <span
              class="DatePicker__calendar__element"
            >
              23
            </span>
            <span
              class="DatePicker__calendar__element"
            >
              24
            </span>
            <span
              class="DatePicker__calendar__element"
            >
              25
            </span>
            <span
              class="DatePicker__calendar__element"
            >
              26
            </span>
            <span
              class="DatePicker__calendar__element"
            >
              27
            </span>
            <span
              class="DatePicker__calendar__element"
            >
              28
            </span>
            <span
              class="DatePicker__calendar__element"
            >
              29
            </span>
            <span
              class="DatePicker__calendar__element"
            >
              30
            </span>
            <span
              class="DatePicker__calendar__element"
            >
              31
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="DatePicker"
    >
      <div
        class="DatePicker__input"
      >
        <div
          class="form-group  "
        >
          <label
            class="col-form-label"
            style="visibility: hidden;"
          >
            Fecha
          </label>
          <div
            class="custom-input-group"
          >
            <input
              class="undefined form-control input-group-br-no"
              type="text"
              value=""
            />
            <span
              class="right-content"
            >
              <div
                class="svg-icon svg-icon-calendar"
              />
            </span>
          </div>
        </div>
      </div>
      <div
        class="DatePicker__calendar"
        tabindex="0"
      >
        <label
          class="col-form-label"
        >
          Selecciona un 
          día
        </label>
        <div
          class="DatePicker__calendar__days-grid"
        >
          <span
            class="DatePicker__calendar__element"
          >
            01
          </span>
          <span
            class="DatePicker__calendar__element"
          >
            02
          </span>
          <span
            class="DatePicker__calendar__element"
          >
            03
          </span>
          <span
            class="DatePicker__calendar__element"
          >
            04
          </span>
          <span
            class="DatePicker__calendar__element"
          >
            05
          </span>
          <span
            class="DatePicker__calendar__element"
          >
            06
          </span>
          <span
            class="DatePicker__calendar__element"
          >
            07
          </span>
          <span
            class="DatePicker__calendar__element"
          >
            08
          </span>
          <span
            class="DatePicker__calendar__element"
          >
            09
          </span>
          <span
            class="DatePicker__calendar__element"
          >
            10
          </span>
          <span
            class="DatePicker__calendar__element"
          >
            11
          </span>
          <span
            class="DatePicker__calendar__element"
          >
            12
          </span>
          <span
            class="DatePicker__calendar__element"
          >
            13
          </span>
          <span
            class="DatePicker__calendar__element"
          >
            14
          </span>
          <span
            class="DatePicker__calendar__element"
          >
            15
          </span>
          <span
            class="DatePicker__calendar__element"
          >
            16
          </span>
          <span
            class="DatePicker__calendar__element"
          >
            17
          </span>
          <span
            class="DatePicker__calendar__element"
          >
            18
          </span>
          <span
            class="DatePicker__calendar__element"
          >
            19
          </span>
          <span
            class="DatePicker__calendar__element"
          >
            20
          </span>
          <span
            class="DatePicker__calendar__element"
          >
            21
          </span>
          <span
            class="DatePicker__calendar__element"
          >
            22
          </span>
          <span
            class="DatePicker__calendar__element"
          >
            23
          </span>
          <span
            class="DatePicker__calendar__element"
          >
            24
          </span>
          <span
            class="DatePicker__calendar__element"
          >
            25
          </span>
          <span
            class="DatePicker__calendar__element"
          >
            26
          </span>
          <span
            class="DatePicker__calendar__element"
          >
            27
          </span>
          <span
            class="DatePicker__calendar__element"
          >
            28
          </span>
          <span
            class="DatePicker__calendar__element"
          >
            29
          </span>
          <span
            class="DatePicker__calendar__element"
          >
            30
          </span>
          <span
            class="DatePicker__calendar__element"
          >
            31
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
